src/main/java/views/add_edit_epense_views/EditExpenseV.java

```diff
@@ -109,7 +106,10 @@ public void actionPerformed(ActionEvent evt) {
         // Set isRecurringCheckBox checked if the expense being edited was a recurring expense
         this.selectedCategory = (String) categoryCombo.getSelectedItem();
         this.isRecurring = isRecurringCheckBox.isSelected();
-        }else if(evt.getSource() == submit){
+        }else if(expenseCombo.getSelectedIndex() == -1){
```

Comment:
I think you would need such a test inside the if ext.getSource() ==submit clause, since when I tried to edit a expense without selecting a expense I want to submit (which shows as if i've selected the first expense in the list), it would just give NoSuchElementException. 

An alternative that I can guarantee it works, is to make the first element of the list a empty 
string "". This way, you can catch the "" case in the if submit clause, and it is also visually more understandable to the user. (This is how I implemented in the MonthMenuView for selecting a month input)

Reply:
I believe I fixed this error and I do agree it's a valid case for a test since it's prone to errors. However, I'm struggling to build a test, any suggestions?

Reply:
You don't build a test for it because it's in the view. You actually run the main() to test it.